bittensor/_axon/__init__.py

```diff
@@ -412,8 +412,12 @@ def parse_signature(
     ) -> Tuple[int, str, str, str, int]:
         r"""Attempts to parse a signature from the metadata"""
         signature = metadata.get("bittensor-signature")
+        version = metadata.get('bittensor-version')
```

Comment:
Metadata is a `Dict[str, str]`. You need to cast to integer if you want to compare it with an integer.

Reply:
Thanks Adrian!